lib/utils.js

```diff
@@ -77,16 +77,33 @@ function waitForAvailable(host, port, options, callback) {
       count--;
       if (available) return callback();
       if (count === 0) return callback(new Error('Server is unavailable'));
-      setTimeout(() => run(), options.retryMs);
+      setTimeout(() => run(), options.retryMS);
     });
   }
 
   run();
 }
 
+function createCommandOptions(manager, localOptions) {
+  const options = Object.assign({}, manager.clientOptions, {
```

Comment:
bah sorry I steered you wrong here, the parameters apply to the left, so I think you want the defaults to be the first object, and your localOptions to be last. I think we need something like:

```
const options = Object.assign({
  host: 'localhost',
  port: '27017',
  pool: 1,
  reconnect: false,
  emitError: true
}, manager.options, localOptions);
```
But please confirm locally first 😄

Reply:
That thought came across my mind but I'm still thinking the current method is good.

First thought, you include `manager.options` & `localOptions` in the assign but I don't think this is a good idea. First, not all of the `manager.options` will apply to the `CoreServer` instance. I think it's best just to extract the host/port from the options and just insert those. Just as an example in the tests, a `dbpath` and `logpath` parameter would be passed to the Server. It probably won't **hurt** anything but I think it's best to only include options that `CoreServer` will use.

Same logic applies to localOptions. Within `executeCommand` there are two options `reExecuteOnError` and `ignoreError` that do not apply to the `CoreServer` instance. In addition, `socketTimeout` is `socketTimeoutMS` in the options.

The current logic mimics what was in the server before. The clientOptions were cloned and then these new options were placed on top.

Reply:
@mbroadst Take a look now. I implemented what you said because I thought about it and I think you're right. The current logic won't have the desired effect at all.

The downside to this approach is that if you set clientOptions & localOptions, clientOptions will win but oh well.

Reply:
@addisonElliott take a look at this [gist](https://gist.github.com/mbroadst/065eba0a73443ee8a0728df3619ace56). I think you should set the defaults in the first object, and let the `clientOptions` override first, then the `localOptions`. As you can see if you run the code snippet, I think we achieve the desired effect.

Reply:
ah I see your concern, I wouldn't worry about the fields that are not used but yes you're correct we should keep those two fields renamed

Reply:
Sorry, I'm a bit confused by your comment. What would you like me to change?

Here is how I think we should do it (your input is welcome). Since at this point we are just passing the options directly to `CoreServer`, how about I change the documentation to tell them to reference `CoreServer` for valid options? For the time being, the user will specify connectionTimeout and socketTimeout.

Then, I can submit a PR to change the name of those parameters on `mongodb-core`.

Reply:
I don't think we should worry about fields that aren't used by the `Server` class, but rather focus on being able to override these values correctly in this project. For the cases where we need property name translation, let's just do that explicitly:

```
'use strict';

function createCommandOptions(manager, localOptions) {
  const options = Object.assign(
    {
      host: 'defaultHost',
      port: 27017,
      connectionTimeoutMS: 5000,
      socketTimeoutMS: 0,
      pool: 1,
      reconnect: false,
      emitError: true
    },
    manager.clientOptions,
    localOptions
  );

  if (options.connectionTimeoutMS) {
    options.connectionTimeout = options.connectionTimeoutMS;
  }

  if (options.socketTimeoutMS) {
    options.socketTimeout = options.socketTimeoutMS;
  }

  return options;
}
```

Alternatively, let's drop the `MS` and go with the name supported in `Server` directly until we can change the property names there.

Reply:
@mbroadst Went for the alternative option. I'm going to submit a PR on `mongodb-core` now so I don't forget